tools/fuzzer.cpp

```diff
@@ -0,0 +1,274 @@
+        double doubleValue;
+        memcpy(&doubleValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(doubleValue));
+        break;
+      }
+      case ADD_UTC_DATE:
+        builder.add(Value(randomGenerator.mt64(), ValueType::UTCDate));
+        break;
+      case ADD_BINARY:
+        builder.add(Value(std::bitset<64>(randomGenerator.mt64()).to_string()));
+        break;
+      case ADD_EXTERNAL: {
+        Builder extBuilder;
+        extBuilder.add(Value(randomGenerator.mt64()));
+        builder.add(Value(static_cast<void const *>(extBuilder.slice().start()), ValueType::External));
+        break;
+      }
+      case ADD_ILLEGAL:
+        builder.add(Value(ValueType::Illegal));
+        break;
+      case ADD_MIN_KEY:
+        builder.add(Value(ValueType::MinKey));
+        break;
+      case ADD_MAX_KEY:
+        builder.add(Value(ValueType::MaxKey));
+      default:
+        break;
+    }
+    break;
+  }
+}
+
+int main(int argc, char* argv[]) {
+  VELOCYPACK_GLOBAL_EXCEPTION_TRY
+    bool isTypeAssigned = false;
+    size_t numIterations = 1;
+    size_t numThreads = 1;
+    Format format = Format::VPACK;
+
+    int i = 1;
+    while (i < argc) {
+      char const *p = argv[i];
+      if (isOption(p, "--help")) {
+        usage(argv);
+        return EXIT_SUCCESS;
+      } else if (isOption(p, "--vpack") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::VPACK;
+      } else if (isOption(p, "--json") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::JSON;
+      } else {
+        int value;
+        std::from_chars(p, p + sizeof(p), value);
+        if (value >= 1) {
+          if (i == 3) { // as threads and iterations are optional arguments, threads must be preceeded by iterations
+            // in the command line, hence, argument 3
+            numThreads = value;
+          } else {
+            numIterations = value;
+          }
+        } else {
+          usage(argv);
+          return EXIT_FAILURE;
+        }
+      }
+      ++i;
+    }
+
+    size_t itsPerThread = numIterations / numThreads;
+    size_t leftoverIts = numIterations % numThreads;
+
+    auto threadCallback = [format](std::stop_token const& stoken, size_t iterations) {
+      try {
+        RandomGenerator randomGenerator;
+        while (iterations-- > 0) {
+          if (stoken.stop_requested()) {
+            return;
+          }
+          Builder builder;
+          generateVelocypack(builder, 0, randomGenerator, format);
+
+          if (format == Format::JSON) {
+            Parser parser;
+            parser.parse(builder.slice().toJson());
+          } else {
+            Validator validator;
+            validator.validate(builder.slice().start(), builder.slice().byteSize());
+          }
+        }
+      } catch (std::exception const& e) {
+        std::cerr << "Program encountered exception on thread execution: " << e.what() << std::endl;
```

Comment:
Ideally one would like to see the broken JSON / VPack, because otherwise it will be hard to figure out _what_ went wrong!